src/protocol/step.rs

```diff
@@ -74,9 +74,14 @@ impl<S: Step + ?Sized> StepNarrow<S> for Descriptive {
             assert!(!s.contains('/'), "The string for a step cannot contain '/'");
         }
 
-        Self {
-            id: self.id.clone() + "/" + step.as_ref(),
+        let mut id = self.id.clone() + "/";
+        #[cfg(feature = "step-trace")]
+        {
+            id += &(std::any::type_name::<S>().to_owned() + "::");
```

Comment:
Note that ":" is not a valid URL character, so this will only work with the in-memory transport, you should add that as a condition.

Reply:
How about "#"?

Reply:
Oh, I don't think that it is worth trying to fix, but it might be worth restricting this feature to use with in-memory to avoid causing explosions.